diff --git a/client/connection/serverListener.ts b/client/connection/serverListener.ts
index 321ad43..39858cb 100644
--- a/client/connection/serverListener.ts
+++ b/client/connection/serverListener.ts
@@ -9,7 +9,7 @@ import { getTree } from '@/data/rootTree'
 
 const config = {
     enableExpensiveUpdateValidation: false,
-    shouldLog: true,
+    shouldLog: false,
 }
 
 function log(...args: unknown[]) {
diff --git a/client/features/battle/components/MoveMenu.tsx b/client/features/battle/components/MoveMenu.tsx
index e4089d1..639e1c4 100644
--- a/client/features/battle/components/MoveMenu.tsx
+++ b/client/features/battle/components/MoveMenu.tsx
@@ -23,7 +23,7 @@ export default function MoveMenu(): JSX.Element {
         // tl('selected move change');
         // @ts-expect-error
         const elapsed = Date.now() - window.startTime
-        console.log(`round trip move change took ${elapsed / 1000} seconds`)
+        // console.log(`round trip move change took ${elapsed / 1000} seconds`)
         const x = selectedMove.get()
         if (x == null) { return }
         setSm(x)
diff --git a/server/gameState/battle/attack.ts b/server/gameState/battle/attack.ts
index 2fdb9cf..dc8ab59 100644
--- a/server/gameState/battle/attack.ts
+++ b/server/gameState/battle/attack.ts
@@ -2,28 +2,60 @@
 import type { AttackData, CharacterMeta, CharacterMove, CharacterStats, CharacterUid, Effect, EffectType, StanceMultiplier, StanceName } from '@shared'
 
 import { getRulebook } from '@/rulebook'
+import { getGameStateCursor } from '@/util'
 
 import { getTransformed, isSpecial } from './specialMoves'
 
 
-export function getCharacterMovesWithDamageRanges(character: CharacterStats & { stance: StanceName }): CharacterMove[] {
-    return character.moves.map(move => {
+type CharacterWithStance = CharacterStats & { stance: StanceName }
 
-        const damageRange = getMoveMultiplierRange(move).map(multiplier => {
-            const damage = Math.round(
-                character.damage * getAttackMultiplier(character) * multiplier
-            )
+export function getCharacterMovesWithDamageRanges(character: CharacterWithStance): CharacterMove[] {
+    return character.moves.map(move => {
 
-            return damage < 1 ? 1 : damage
-        })
+        const damageRange = getMoveMultiplierRange(move)
+            .map(multiplier => getDamage(character, multiplier))
+            .map(damage => getBlessingAdjustedDamage(character, damage))
 
         return {
             ...move,
             damageRange,
         }
     })
+
+    function getDamage(character: CharacterStats, multiplier: number): number {
+
+        const damage = Math.round(
+            character.damage * getAttackMultiplier(character) * multiplier
+        )
+
+        return damage < 1 ? 1 : damage
+
+    }
+
+
 }
 
+function getBlessingAdjustedDamage(character: CharacterStats, damage: number): number {
+    const bls = getGameStateCursor('alice').get('blessings')
+    const type = character.isPc ? 'party' : 'enemies'
+    debugger
+    for (const b of bls) {
+        for (const ef of b.effects) {
+            if (ef.target === type
+                || typeof ef.target === 'object' && ef.target.type === type && ef.target.characterType === character.name) {
+                // console.log('you have an effect affecting you')
+                console.log(`Character ${character.name} has effect ${b.name}`)
+
+                const mult = ef.damageMultiplicand ?? 1
+                const add = ef.damageAddend ?? 0
+                damage = damage * mult + add
+            }
+        }
+    }
+    return damage
+}
+
+
 export function getCharacterKeysAndDamages(attackData: AttackData): { key: CharacterUid, damage: number }[] {
     const kds = attackData.defenders.map(defender => (
         { key: defender.uid, damage: getDamage(attackData, defender) }
diff --git a/server/rulebook/blessingsMap.ts b/server/rulebook/blessingsMap.ts
index 45464a2..0f64311 100644
--- a/server/rulebook/blessingsMap.ts
+++ b/server/rulebook/blessingsMap.ts
@@ -25,7 +25,8 @@ export const blessingsMap: Record<BlessingName, Blessing> = {
         ],
     },
     'ptbotflax': {
-        name: 'Play The Bones of The Fallen Like a Xylophone',
+        name: 'ptbotflax',
+        displayName: 'Play The Bones of The Fallen Like a Xylophone',
         after: { doorType: 'skeleton' },
         effects: [
             {
diff --git a/shared/datamodel.ts b/shared/datamodel.ts
index d043c2c..8424b8a 100644
--- a/shared/datamodel.ts
+++ b/shared/datamodel.ts
@@ -118,6 +118,7 @@ type TargetType = 'party' | 'enemies'
 
 export type Blessing = {
     name: string,
+    displayName?: string,
     after?: {
         doorType?: SpecialDoorName
     }
