diff --git a/client/features/gather/elements/Gather.ts b/client/features/gather/elements/Gather.ts
index a6e808e..f18509e 100644
--- a/client/features/gather/elements/Gather.ts
+++ b/client/features/gather/elements/Gather.ts
@@ -1,8 +1,36 @@
+import { Button, FastLayoutOptions, Stage as UXStage, TextInput } from '/Users/l/git/qpwo-pixi-ui/packages/core/build' // '@puxi/core'
 import type { PixiContainer } from '@/elementsUtil'
-import { Container, PixiTexture, Sprite } from '@/elementsUtil'
-
+import { Container } from '@/elementsUtil'
 export function Gather(): PixiContainer {
+    const uxStage = new UXStage(512, 512)
+    uxStage.addChild(
+        new Button({
+            text: 'Hello world!',
+        })
+    )
+
+    uxStage.addChild(
+        new TextInput({
+            value: 'Click me!',
+        })
+            .setLayoutOptions(
+                new FastLayoutOptions({
+                    width: 500, // LayoutOptions.WRAP_CONTENT, // width
+                    height: 60, // height
+                    x: 0.5,
+                    y: 0.5, // x, y (center)
+                    // anchor: 0.5, //FastLayoutOptions.CENTER_ANCHOR, // properly center
+                })
+            )
+            .setPadding(4, 6) // horizontal/vertical padding
+            .setBackground(0xffffff) // background color (can use a PIXI.Graphics too)
+            .setBackgroundAlpha(0.9) // alpha for background
+            .setElevation(2) // drop-shadow on background!
+    )
     return Container({
+        children: [uxStage],
+    })
+    /* return Container({
         children: [
             Sprite({
                 src: PixiTexture.WHITE,
@@ -13,5 +41,5 @@ export function Gather(): PixiContainer {
                 y: 200,
             }),
         ],
-    })
+    })*/
 }
