diff --git a/server/auth/regex.spec.ts b/server/auth/regex.spec.ts
new file mode 100644
index 0000000..b728e2c
--- /dev/null
+++ b/server/auth/regex.spec.ts
@@ -0,0 +1,32 @@
+import { expect, Test, TestSuite } from 'testyts'
+
+import { isValidUsername } from './util'
+
+@TestSuite()
+export class RegexTestSuite {
+    @Test()
+    usernameRegexTest() {
+        const cases: [string, boolean][] = [
+            ['a', false],
+            ['abc', true],
+            ['abc-d', true],
+            ['abc_d', false],
+            ['🎉', false],
+            ['abra-ibra_cadabra.2', false],
+            ['a'.repeat(24), true],
+            ['a'.repeat(25), true],
+            ['a'.repeat(26), false],
+            ['!abra-ibra', false],
+            ['(abra)_ibra', false],
+        ]
+        for (const [input, expectedOutput] of cases) {
+            expect.toBeEqual(
+                isValidUsername(input),
+                expectedOutput,
+                `username ${input} had output ${isValidUsername(
+                    input
+                )}, but should've been ${expectedOutput}`
+            )
+        }
+    }
+}
diff --git a/server/auth/util.ts b/server/auth/util.ts
new file mode 100644
index 0000000..a5c7cb9
--- /dev/null
+++ b/server/auth/util.ts
@@ -0,0 +1,4 @@
+const usernamePattern = /^([A-Z]|[a-z]|-){3,25}$/
+export function isValidUsername(username: string): boolean {
+    return usernamePattern.test(username)
+}
